charts/newrelic-k8s-metrics-adapter/templates/apiservice/job-patch/job-createSecret.yaml

```diff
@@ -3,48 +3,56 @@ apiVersion: batch/v1
   labels:
-    app: {{ template "newrelic-k8s-metrics-adapter.name" $ }}
-{{ include "newrelic-k8s-metrics-adapter.labels" $ | indent 4 }}
+    app: {{ template "common.naming.name" $ }}
+{{ include "common.labels" $ | indent 4 }}
 spec:
   template:
     metadata:
-      name: {{ template "newrelic-k8s-metrics-adapter.fullname" . }}-apiservice-create
+      name: {{ template "common.naming.fullname" . }}-apiservice-create
       labels:
-        app: {{ template "newrelic-k8s-metrics-adapter.name" $ }}
-{{ include "newrelic-k8s-metrics-adapter.labels" $ | indent 8 }}
+        app: {{ template "common.naming.name" $ }}
+{{ include "common.labels" $ | indent 8 }}
     spec:
       containers:
         - name: create
-          image: {{ .Values.apiServicePatchJob.image.repository }}:{{ .Values.apiServicePatchJob.image.tag }}
+          image: {{ include "common.images.image" ( dict "imageRoot" .Values.apiServicePatchJob.image "context" .) }}
           imagePullPolicy: {{ .Values.apiServicePatchJob.image.pullPolicy }}
           args:
             - create
-            - --host={{ template "newrelic-k8s-metrics-adapter.fullname" . }},{{ template "newrelic-k8s-metrics-adapter.fullname" . }}.{{ .Release.Namespace }}.svc
+            - --host={{ template "common.naming.fullname" . }},{{ template "common.naming.fullname" . }}.{{ .Release.Namespace }}.svc
             - --namespace={{ .Release.Namespace }}
-            - --secret-name={{ template "newrelic-k8s-metrics-adapter.fullname" . }}-apiservice
+            - --secret-name={{ template "common.naming.fullname" . }}-apiservice
             - --cert-name=tls.crt
             - --key-name=tls.key
-          {{- if .Values.apiServicePatchJob.image.volumeMounts }}
+          {{- with .Values.apiServicePatchJob.extraEnv }}
+          env:
+          {{- toYaml . | nindent 10 }}
+          {{- end }}
+          {{- with .Values.apiServicePatchJob.extraEnvFrom }}
+          envFrom:
+          {{- toYaml . | nindent 10 }}
+          {{- end }}
+          {{- with .Values.apiServicePatchJob.extraVolumeMounts }}
           volumeMounts:
-          {{- include "tplvalues.render" ( dict "value" .Values.apiServicePatchJob.image.volumeMounts "context" $ ) | nindent 10 }}
+          {{- toYaml . | nindent 10 }}
           {{- end }}
-      {{- if .Values.apiServicePatchJob.image.volumes }}
+      {{- with .Values.apiServicePatchJob.extraVolumes }}
       volumes:
-      {{- include "tplvalues.render" ( dict "value" .Values.apiServicePatchJob.image.volumes "context" $ ) | nindent 8 }}
+      {{- toYaml . | nindent 6 }}
       {{- end }}
       restartPolicy: OnFailure
-      serviceAccountName: {{ template "newrelic-k8s-metrics-adapter.fullname" . }}-apiservice
+      serviceAccountName: {{ include "common.serviceAccount.name" . }}-apiservice
       securityContext:
         runAsGroup: 2000
         runAsNonRoot: true
         runAsUser: 2000
```

Comment:
Pending PR from the common-library.

Reply:
This chart has 2 different podSecurityContexts, the one in the deployment is coming from the values using the helper, but this one we have not been letting the users modify it and can't be in the values with the format expected by the helper since it's already used by the deployment one.

Reply:
No idea if it would make sense to have an other value for that..  😕 
Maybe we can keep that fixed for now, no strong opinion

Using the "same" security context could lead to issues if at some point we need them to be different

Reply:
I agree, I would leave it non modifiable in this case, WDYT Juanma?

Reply:
oks, I keep it fixed for now